src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -1025,6 +1065,221 @@ private static string GetFirstLineSubString(string stringToComplete, out bool ha
+
+            if (variableAst == null)
+            {
+                return false;
+            }
+
+            var variableValue = completionContext.ExecutionContext.GetVariableValue(variableAst.VariablePath);
+            Type type = variableValue.GetType();
+
+            // Assignment constraints override any existing ones, so try them first
+
+            // Check any [ValidateSet()] constraint first since it's likely to be narrow
+            if (setConstraint?.ValidValues != null)
+            {
+                completions = GetResultForSet(type, setConstraint.ValidValues, completionContext);
+                return true;
+            }
+
+            // Then try to complete for an enum type
+            if (variableValue != null)
+            {
+                if (type.IsEnum)
+                {
+                    completions = GetResultForEnum(type, completionContext);
+                    return true;
+                }
+            }
+
+            // If the assignment itself was unconstrained, the variable still might be
+            if (!TryGetTypeConstraintOnVariable(completionContext, variableAst.VariablePath.UserPath, out typeConstraint, out setConstraint))
+            {
+                return false;
+            }
+
+            // Again try the [ValidateSet()] constraint first
+            if (setConstraint?.ValidValues != null)
+            {
+                completions = GetResultForSet(type, setConstraint.ValidValues, completionContext);
+                return true;
+            }
+
+            // Then try to complete for an enum type
+            if (!typeConstraint.IsEnum)
+            {
+                return false;
+            }
+            else
+            {
+                completions = GetResultForEnum(typeConstraint, completionContext);
+                return true;
+            }
+        }
+
+        private static List<CompletionResult> GetResultForSet(
+            Type type,
+            IList<string> validValues,
+            CompletionContext completionContext)
+        {
+            var allValues = new List<string>();
+            foreach (string value in validValues)
+            {
+                if (type == typeof(string) || type.IsEnum)
+                {
+                    allValues.Add(GetQuotedString(value, completionContext));
+                }
+                else
+                {
+                    allValues.Add(value);
+                }
+            }
+
+            return GetMatchedResults(allValues, completionContext);
+        }
+
+        private static List<CompletionResult> GetMatchedResults(
+            List<string> allValues,
+            CompletionContext completionContext)
+        {
+            var stringToComplete = string.Empty;
+            if (completionContext.TokenAtCursor != null)
+            {
+                stringToComplete = completionContext.TokenAtCursor.Text;
```

Comment:
I think you can use `completionContext.WordToComplete`.

Reply:
ok

Reply:
This doesn't work.  WordToComplete is null.